opentelemetry/src/main/java/io/grpc/opentelemetry/MetadataSetter.java

```diff
@@ -57,6 +60,8 @@ public void set(@Nullable Metadata carrier, String key, String value) {
   }
 
   void set(@Nullable Metadata carrier, String key, byte[] value) {
+    logger.log(Level.FINE, "Setting binary trace header key={0} value={1}",
+        new Object[]{key, Arrays.toString(value)});
```

Comment:
This toString() is performed even when logging is disabled. Does `Arrays.asList()` work?